prow/github/client.go

```diff
@@ -63,6 +64,99 @@ const (
 	initialDelay  = 2 * time.Second
 )
 
+// Interface for how prow interacts with the http client, which we may throttle.
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Inteface for how prow interacts with the graphql client, which we may throttle.
+type gqlClient interface {
+	Query(ctx context.Context, q interface{}, vars map[string]interface{}) error
+}
+
+// throttler sets a ceiling on the rate of github requests.
+// Configure with Client.Throttle()
+type throttler struct {
+	ticker   *time.Ticker
+	throttle chan time.Time
+	Http     httpClient
+	Graph    gqlClient
```

Comment:
Do `Http` and `Graph` need to be exposed outside the package?

Reply:
Done